Packs/DragosWorldview/Integrations/DragosWorldview/DragosWorldview.py

```diff
@@ -79,26 +76,57 @@ def get_stix(client: Client, args: Dict[str, Any]) -> Dict[str, Any]:
 
 
 def get_indicators(client: Client, args: Dict[str, Any]) -> CommandResults:
-    serial = args.get('serial')
-    if serial:
-        api_query = "indicators?serial%5B%5D=" + serial
+    exclude_suspect_domain = args.get('exclude_suspect_domain')
+    page = args.get('page')
+    page_size = args.get('page_size')
+    updated_after = args.get('updated_after')
+    value = args.get('value')
+    indicator_type = args.get('type')
+    serials = argToList(args.get('serial'))
+    tags = argToList(args.get('tags'))
+
+    # The arguments page, page_size and exclude_suspect_domain have an API default of 1, 500 and false respectively,
+    # and do not need to be included in the query unless changed
+    query_list = []
+    if exclude_suspect_domain:
+        query_list.append('exclude_suspect_domain=' + exclude_suspect_domain)
+    if page:
+        query_list.append('page=' + page)
+    if page_size:
+        query_list.append('page_size=' + page_size)
+    if updated_after:
+        query_list.append('updated_after=' + updated_after.replace(":", "%3A"))
+    if value:
+        query_list.append('value=' + value)
+    if indicator_type:
+        query_list.append('type=' + indicator_type)
+    for serial in serials:
+        query_list.append('serial%5B%5D=' + serial)
+    for tag in tags:
+        query_list.append('tags%5B%5D=' + tag)
+
+    # If any arguments were submitted run the relevent query, else run the default
+    # behavior (returning all indicators from the last 48 hours)
+    if query_list:
+        query_string = '&'.join(query_list)
+        api_query = f'indicators?{query_string}'
     else:
-        time = datetime.now() - timedelta(hours=48)
-        api_query = "indicators?updated_after="
-        api_query = api_query + str(time)
-        api_query = api_query.replace(":", "%3A")
-
+        time = str(datetime.now() - timedelta(hours=48))
+        time = time.replace(":", "%3A")
+        api_query = f'indicators?updated_after={time}'
     raw_response = client.api_request(api_query)
     data = raw_response['indicators']
-    page_number = 2
+    page_number = 2 if not page else page + 1
     full_response = raw_response
 
-    while raw_response['total_pages'] >= raw_response['page']:
-        if serial:
-            api_query = "indicators?page=" + str(page_number) + "&serial%5B%5D=" + serial
+    # If there are still more dragos pages (ie more indicators) than was returned by
+    # the intial query, iterate through the remaining pages and add all unique indicators
+    # to the return data
+    while raw_response['total_pages'] > raw_response['page']:
+        if query_list:
+            api_query = f'indicators?page={page_number}&{query_string}'
```

Comment:
please note that the `query_string` may already contained `page` from the previous iteration 
so the `api_query` will may have two different `page` value

Reply:
I've added a new check in lines 120-122 which will remove the old `page` value and recreate `query_string` without it

Reply:
@NicCas the only point remain to check here is if `raw_response['total_pages']` and `raw_response['page']` are numbers, because if they are `str` this may cause issues as `"2" > "10"`

Reply:
Good point! I've added type casting to that while check